File: java/05 - Lab 3 - PetShop/src/main/java/org/cegeka/petshop/Shop.java
package org.cegeka.petshop;

import com.google.common.collect.ImmutableSet;

import java.util.*;

public class Shop {

    private Set<String> itemList = new HashSet<String>();

    public void addItem(String item) {
        itemList.add(item);
    }

    public Set<String> getItemsInStock() {
        return ImmutableSet.copyOf(itemList);
    }
}

File: csharp/05 - Lab 3 - PetShop/PetShop/Shop.cs
using System.Collections.Generic;
using System.Collections.Immutable;

namespace PetShop
{
    public class Shop
    {
        private ISet<string> _itemList = new HashSet<string>();

        public void AddItem(string item) => _itemList.Add(item);  

        public ISet<string> ItemsInShop => _itemList.ToImmutableHashSet();
    }
}
